module/executiondatasync/optimistic_sync/persisters/stores/transaction_result_error_messages.go

```diff
@@ -39,10 +42,12 @@ func (t *TxResultErrMsgStore) Persist(lctx lockctx.Proof, batch storage.ReaderBa
 		return fmt.Errorf("could not get transaction result error messages: %w", err)
 	}
 
-	if len(txResultErrMsgs) > 0 {
-		if err := t.persistedTxResultErrMsg.BatchStore(t.blockID, txResultErrMsgs, batch); err != nil {
-			return fmt.Errorf("could not add transaction result error messages to batch: %w", err)
-		}
+	err = storage.SkipAlreadyExistsError( // Note: if the data already exists, we will not overwrite
```

Comment:
@peterargue could you confirm this behavior?

I'm not sure if in the past, we let AN overwrite any data in the scenario of recovery or backfilling. 

But the current behavior now is that it will never overwrite, is that OK?